applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_3d.py

```diff
@@ -0,0 +1,351 @@
+import KratosMultiphysics
+import KratosMultiphysics.CompressiblePotentialFlowApplication as CPFApp
+import math
+
+
+def DotProduct(A,B):
+    result = 0
+    for i,j in zip(A,B):
+        result += i*j
+    return result
+
+def Factory(settings, Model):
+    if(not isinstance(settings, KratosMultiphysics.Parameters)):
+        raise Exception(
+            "expected input shall be a Parameters object, encapsulating a json string")
+
+    return DefineWakeProcess3D(Model, settings["Parameters"])
+
+class DefineWakeProcess3D(KratosMultiphysics.Process):
+    def __init__(self, Model, settings):
+        KratosMultiphysics.Process.__init__(self)
+
+        # Check default settings
+        default_settings = KratosMultiphysics.Parameters(r'''{
+            "model_part_name": "",
+            "body_model_part_name": "",
+            "wake_stl_file_name" : "",
+            "wake_normal": [0.0,0.0,1.0],
+            "output_wake": true,
+            "epsilon": 1e-9
+        }''')
+        settings.ValidateAndAssignDefaults(default_settings)
+
+        self.model = Model
+
+        trailing_edge_model_part_name = settings["model_part_name"].GetString()
+        if trailing_edge_model_part_name == "":
+            err_msg = "Empty model_part_name in DefineWakeProcess3D\n"
+            err_msg += "Please specify the model part that contains the trailing edge nodes"
+            raise Exception(err_msg)
+        self.trailing_edge_model_part = Model[trailing_edge_model_part_name]
+
+        body_model_part_name = settings["body_model_part_name"].GetString()
+        if body_model_part_name == "":
+            err_msg = "Empty model_part_name in DefineWakeProcess3D\n"
+            err_msg += "Please specify the model part that contains the body nodes"
+            raise Exception(err_msg)
+        self.body_model_part = Model[body_model_part_name]
+
+        self.wake_stl_file_name = settings["wake_stl_file_name"].GetString()
+        if self.wake_stl_file_name == "":
+            err_msg = "Empty wake_stl_file_name in DefineWakeProcess3D\n"
+            err_msg += "Please specify the stl file name that contains the wake surface nodes"
+            raise Exception(err_msg)
+
+        self.epsilon = settings["epsilon"].GetDouble()
+        self.output_wake = settings["output_wake"].GetBool()
+
+        # For now plane wake surfaces are considered.
+        # TODO: Generalize this to curved wake surfaces
+        self.wake_normal = settings["wake_normal"].GetVector()
```

Comment:
Good point. For a wing with a straight trailing edge we can compute the wake_normal using the wake direction and the span direction. I didn't implement this because I thought that anyway the goal is to generalize this for curved trailing edges.

For a wing with a curved trailing edge the wake normal will not be unique and we will need to compute the normal in each discretized triangle wake surface element and then pass this information to the cut fluid wake element to apply the wake condition.

Do you need this feature? If you want we can automatize it now for straight trailing edges and then see what we do for curved trailnig edges when the time comes.